src/com/amazon/corretto/crypto/provider/AesGcmSpi.java

```diff
@@ -916,10 +905,8 @@ protected int engineUpdate(ByteBuffer input, final ByteBuffer output)
           throw new ShortBufferException();
         }
 
-        if (Utils.buffersMaybeOverlap(input, output)) {
-          // We'll just copy the whole input buffer if it might overlap with output. It's possible
-          // to do something more efficient for a couple of special cases here, but we'll do the
-          // simple and safe thing for now.
+        if (Utils.outputClobbersInput(input, output)) {
+          // We'll just copy the whole input buffer if it might overlap with output.
           ByteBuffer newInput = ByteBuffer.allocate(input.remaining());
```

Comment:
This is where we get our memory savings. That ByteBuffer allocation is now avoided for the common case of overlapping buffers.